enterprise/cmd/frontend/internal/registry/allow.go

```diff
@@ -56,3 +77,15 @@ func getAllowedExtensionsFromSiteConfig() []string {
 	}
 	return nil
 }
+
+func getAllowOnlySourcegraphAuthoredExtensionsFromSiteConfig() bool {
+	if c := conf.Get().Extensions; c != nil && (c.RemoteRegistry == nil || c.RemoteRegistry == conf.DefaultRemoteRegistry) {
+		return c.AllowOnlySourcegraphAuthoredExtensions
+	}
+
+	return false
+}
+
+func isSourcegraphAuthoredExtension(x *registry.Extension) bool {
+	return x.Publisher.Name == "sourcegraph"
+}
```

Comment:
nit:
```suggestion
func isSourcegraphAuthoredExtension(p *client.Publisher) bool {
	return p.Name == "sourcegraph"
}
```

All we need is the publisher, doesn't seem worth passing in the entire extension struct.